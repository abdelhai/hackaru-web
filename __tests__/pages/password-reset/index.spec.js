import { Store } from 'vuex-mock-store';
import { shallowMount } from '@vue/test-utils';
import Index from '@/pages/password-reset/index';

describe('Index', () => {
  let wrapper;

  const $store = new Store({});
  const factory = () =>
    shallowMount(Index, {
      mocks: {
        $store
      }
    });

  beforeEach(() => {
    $store.reset();
  });

  describe('when click submit-button', () => {
    beforeEach(() => {
      wrapper = factory();
      wrapper.find('.email').vm.$emit('input', 'example@example.com');
      wrapper.find('form').trigger('submit.prevent');
    });

    it('dispatch auth/sendPasswordResetEmail', () => {
      expect($store.dispatch).toHaveBeenCalledWith(
        'auth/sendPasswordResetEmail',
        {
          email: 'example@example.com',
          redirectUrl: 'http://localhost/en/password-reset/edit'
        }
      );
    });
  });
});
