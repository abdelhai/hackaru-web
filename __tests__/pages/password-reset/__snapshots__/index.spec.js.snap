// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index render correctly 1`] = `
<section>
  <heading-stub
    class="is-large"
  >
    
    title
  
  </heading-stub>
   
  <p>
    about
  </p>
   
  <form>
    <text-field-stub
      aria-label="email"
      autofocus=""
      class="has-border"
      placeholder="email"
      required=""
      type="email"
      value=""
    />
     
    <footer>
      <btn-stub
        class="is-rounded is-primary"
        type="submit"
      >
        
        send
      
      </btn-stub>
    </footer>
  </form>
</section>
`;
